lib/game.rb

```diff
@@ -0,0 +1,193 @@
+require 'pry'
+class Game
+	WIN_COMBINATIONS = [
+											[0,1,2],
+											[3,4,5],
+											[6,7,8],
+											[0,3,6],
+											[1,4,7],
+											[2,5,8],
+											[0,4,8],
+											[6,4,2]
+											]
+	DIVIDER = "===================\n"									
+											
+	attr_accessor :player_1, :player_2, :board, :war_mode, :war_stats
+	attr_reader :the_winner, :complete
+	
+	def initialize(player_1=Human.new("X"), player_2=Human.new("O"), board=Board.new)
+		@player_1 = player_1
+		@player_2 = player_2
+		@board = board
+		@complete = false
+		@war_mode = false
+		@war_stats = {
+			:num_games => 0,
+			:x_wins => 0,
+			:o_wins => 0,
+			:draws => 0
+		}
+	end
+	
+	def current_player
+		board.turn_count.even? ? player_1 : player_2
+	end
+	
+	def over?
+		won? ? true : draw?
+	end
+	
+	def completed?
+		war_mode ? @complete : over?
+	end
+	
+	def won?
+		WIN_COMBINATIONS.any? do |combo|
+			all_same?(combo, "X") || all_same?(combo, "O")
+		end
+	end
+	
+	def all_same?(array, token)
+		if array.all? {|i| board.cells[i] == token}
+			@the_winner = token
+			true
+		else
+			@the_winner = nil
+			false
+		end
+	end
+	
+	def draw?
+		!board.full? ? false : !won?
+	end
+	
+	def winner
+		won? ? the_winner : nil
```

Comment:
When you see the pattern of a ternary operator returning `nil`, you can replace it with an inline if statement, which has the same effect.
`the_winner if won?`